### PR TITLE
[ADD] portal_sale_personal_data_only: New module

### PR DESCRIPTION
**Pull request by chienandalu**
a member of their organization are followers. That could cause a leaking of
documents between members and departments and of the organization that should
stay private.

This module restricts that behaviour so the portal users only see their own
documents.

@Tecnativa

### CODE REVIEW

**Review by yajo — APPROVED** [review]
I'd add to the roadmap support for per-company policy.

The rest seems OK.

**Comment by JayVora-SerpentCS** [comment]
LGTM

**Comment by pedrobaeza** [comment]
Please attend runbot warning:
```
2018-09-28 08:27:20,651 200 WARNING openerp_test odoo.addons.base.module.module: module portal_sale_personal_data_only: description is empty !
```

**Comment by JayVora-SerpentCS** [comment]
Can you please check travis and runbot failures ?

**Comment by LoisRForgeFlow** [comment]
ping @chienandalu

**Comment by chienandalu** [comment]
Please, review again

File: portal_sale_personal_data_only/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+By default, portal users are allowed to see al the orders and invoices in wich
```
Comment (yajo) [inline portal_sale_personal_data_only/readme/DESCRIPTION.rst]: "all", "which"

**Review by yajo — APPROVED** [review]

**Comment by chienandalu** [comment]
I just updated this PR to fix some things that weren't correct:

- The new record rule had no effect as it was a more restrictive rule and Odoo combines them with an `OR` operator. So the portal_sale rules domains have to be overriden.
- The rule is applied to the descendants of the partner, so partner A can see any document of their children contacts.
- A partner can see an invoice with lines related to a sale order he owns.

A final doubt I've got is what to do when the module is uninstalled, options being:

- Nothing
- Restore original rule
- Restore original rule only if the rule wasn't modified later.

**Comment by yajo** [comment]
This is a difficult case indeed...

You should possibly do:

- On install:
  - Deactivate upstream rules that conflict.
  - Add yours.
- On uninstall:
  - Reactivate upstream rules.
  - Remove yours (this should be automatic).

You'll have to keep in mind that these records are noupdate (if I'm not wrong), so maybe you'll have to do those modifications in a script instead than a data file.

**Comment by chienandalu** [comment]
That's it, @Yajo :) Changes done

**Comment by yajo** [comment]
Nice! Now just check the bots please.

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍 

Useful module!

**Comment by chienandalu** [comment]
Well, looks like CI is broken for this branch :confused:

**Review by rafaelbn — APPROVED** [review]
:+1:

**Comment by rafaelbn** [comment]
Please @chienandalu review travis and runbot :red_circle:

**Comment by rafaelbn** [comment]
Please @chienandalu review bots :red_circle:

**Comment by chienandalu** [comment]
This branch was a little bit broke. I just rebased to see how behaves now

**Comment by chienandalu** [comment]
Nope, seems that Travis can't boot properly in this branch

**Comment by yajo** [comment]
Add postgresql: 9.6 to addons in .travis.yml.

**Comment by rafaelbn** [comment]
Great @chienandalu ! Travis green :green_heart: 

Just missing tests:  codecov/patch — 61.53% of diff hit (target 82.67%)

**Pull request merged**